tests/test_operation.py

```diff
@@ -836,6 +836,15 @@ def parameter_frequencies(self):
         for i in range(num_param):
             assert f[i] == (0.2, x[i])
 
+    def test_frequencies_sparse_generator(self):
+        """Test that the parameter frequencies are correctly deduced from a generator
+        that is a ``SparseHamiltonian``."""
+        DummyOp = copy.copy(qml.DoubleExcitationPlus)
+        DummyOp.parameter_frequencies = qml.operation.Operation.parameter_frequencies
+
+        op = DummyOp(0.7, [0, 1, 2, 3])
+        assert op.parameter_frequencies == [(1.0,)]
```

Comment:
Could we add something like this, just to confirm the test continues to check a case where the generator is a `SparseHamiltonian` (even if we change things later)?

```suggestion
        assert isinstance(op.generator(), qml.SparseHamiltonian)
        assert op.parameter_frequencies == [(1.0,)]
```